mithril-client/src/certificate_client/verify_cache/memory_cache.rs

```diff
@@ -0,0 +1,337 @@
+use async_trait::async_trait;
+use chrono::{DateTime, TimeDelta, Utc};
+use std::collections::HashMap;
+use std::ops::Add;
+use tokio::sync::RwLock;
+
+use crate::certificate_client::CertificateVerifierCache;
+use crate::MithrilResult;
+
+pub type CertificateHash = str;
+pub type PreviousCertificateHash = str;
+
+/// A in-memory cache for the certificate verifier.
+pub struct MemoryCertificateVerifierCache {
+    expiration_delay: TimeDelta,
+    cache: RwLock<HashMap<String, CachedCertificate>>,
```

Comment:
The defined alias isn't on `String` but on `str`, that's why I did not reuse it here :sweat_smile:, I could add another alias but this would add confusion with the existing alias on `str`.

Reply:
Alright👌! 
And what do you think about renaming `CachedCertificate` to `CachedParentCertificate`?

Reply:
I would rather not, I think that keeping it as a `CachedCertificate` with a `parent_hash` field would sustain changes more easily.
For example if we added the avk inside this structure that would be the avk of the certificate, not of its parent, so this naming would conflict with the new intent that we would like to express.

Reply:
Okay, that works for me. Thanks for the explanation 😊.